util/template/client-class

```diff
@@ -60,19 +67,40 @@ class Client
     /**
      * Client constructor
      *
-     * @param Transport                   $transport
-     * @param callable                    $endpoint
+     * @param Transport $transport
+     * @param callable|EndpointFactoryInterface $endpointFactory
      * @param NamespaceBuilderInterface[] $registeredNamespaces
      */
-    public function __construct(Transport $transport, callable $endpoint, array $registeredNamespaces)
+    public function __construct(Transport $transport, callable|EndpointFactoryInterface $endpointFactory, array $registeredNamespaces)
     {
         $this->transport = $transport;
-        $this->endpoints = $endpoint;
+        if (is_callable($endpointFactory)) {
+            @trigger_error('Passing a callable as the $endpointFactory param in ' . __METHOD__ . ' is deprecated in 2.3.2 and will be removed in 3.0.0. Pass an instance of \OpenSearch\EndpointFactoryInterface instead.', E_USER_DEPRECATED);
+            $endpoints = $endpointFactory;
+            $endpointFactory = new LegacyEndpointFactory($endpointFactory);
+        } else {
+            $endpoints = $endpointFactory->getEndpoint(...);
```

Comment:
this is wrong because `$endpoints` is not expecting a class-string but a substring of it without the base namespace.

Reply:
```suggestion
            $endpoints = fn ($c) => $endpointFactory->getEndpoint('OpenSearch\\Endpoints\\' . $c);
```